lib/spack/spack/compiler.py

```diff
@@ -592,7 +595,16 @@ def search_regexps(cls, language):
         # defined for the compiler
         compiler_names = getattr(cls, "{0}_names".format(language))
         prefixes = [""] + cls.prefixes
-        suffixes = [""] + cls.suffixes
+        suffixes = [""]
+        # Windows compilers generally have an extension of some sort
+        # as do most files on Windows, handle that case here
+        if is_windows:
+            ext = r"\.(exe|bat)+"
```

Comment:
What is the purpose of `+` in this regex? My interpretation is that it would match `batbat` or `exebatexeexe` but that doesn't seem useful.

Reply:
Cruft, It's effectively a noop unless someone truly wants to see what they can get away with in terms of extension names. I'll drop it out.